### ISSUE TITLE
3.2.0 release planning

### ISSUE THREAD

**Issue by mr-c** [OpenMS/OpenMS#6953]
Target date: 2nd week of November, 2023

Yesterday I told @cbielow that I would volunteer to be the release manager for the next release

https://github.com/OpenMS/OpenMS/wiki/Preparation-of-a-new-OpenMS-release#release_developer

https://github.com/OpenMS/OpenMS/wiki/Testing-an-OpenMS-release-candidate

→ Release checklist is at https://github.com/OpenMS/OpenMS/issues/7123

Goals for the next release:
- [x] Gather fixes for issues reported against 3.0.0
   - [x] https://github.com/OpenMS/OpenMS/issues/6958
- [x] Fixed JSON loading for CWL support: https://github.com/OpenMS/OpenMS/pull/6950
- [ ] CWL generation: https://github.com/OpenMS/OpenMS/pull/6156 & documentation
  - [x] List CWL under ["Recommended Workflow Systems"](https://openms.readthedocs.io/en/latest/run-workflows-with-openms-tools/recommended-workflow-systems.html)
  - [ ] Fix [workflow download list](https://openms.readthedocs.io/en/latest/downloads.html#workflows); maybe replace that section with workflow-language specific links into <https://workflowhub.org> ?
  - [ ] Ensure that CWL tool descriptions are in the binary release archives
  - [ ] Publish CWL tool descriptions to <https://github.com/common-workflow-library/bio-cwl-tools> or other community repository

**Comment by timosachsenberg** [comment]
Do CWL tool descriptions really need to be in the source archives?

**Comment by mr-c** [comment]
> Do CWL tool descriptions really need to be in the source archives?

Good catch, I removed that from the checklist.

**Comment by mr-c** [comment]
@timosachsenberg Can you make a `release/3.0.1` branch so that I can open PRs?

**Comment by jpfeuffer** [comment]
@mr-c We usually do development on `develop` and branch off once a release is imminent. Once we branch off, release candidates start to be built. They are then tested and declared final/tagged once manual tests are successful.
I would say we are not married to that strategy but unless there are convincing arguments to change it, it is less work to continue doing so.
Does that sound like a reasonable strategy to you, too?

**Comment by jpfeuffer** [comment]
Thinking about it, it would actually work to start branching off now to make sure only certain controlled PRs go into the next release. In the end it is a similar strategy. It's just some more jobs/tests to supervise at the same time for a longer period of time.

**Comment by mr-c** [comment]
@jpfeuffer Is the goal "3.0.1" (3.0 plus some small fixes, but no new features) or "3.1.0" (some new bits + fixes)?

If the goal is the first, then I suggest a `release/3.0.1` branch off of https://github.com/OpenMS/OpenMS/tree/Release3.0.0 to keep it clean. Then each change we want in the `3.0.1` release can be cherry-picked in.

Otherwise, working off of `develop` is fine by me.

**Comment by jpfeuffer** [comment]
Good point. I would be fine with using everything accumulated so far, including new smaller features and do a 3.1. I think  the new features are so minor that they won't impact stability negatively.

**Comment by poshul** [comment]
@mr-c we'd be happy to have you as release manager for 3.1.0. Do you want to set up a meeting to figure out what that's gonna entail?

**Comment by mr-c** [comment]
@poshul Great, thanks! Yes, I'll send you an email with my availability.
